crates/bevy_app/src/app.rs

```diff
@@ -475,17 +512,38 @@ impl App {
     ///     .insert_non_send_resource(MyCounter { counter: 0 });
     /// ```
     pub fn insert_non_send_resource<R: 'static>(&mut self, resource: R) -> &mut Self {
-        self.world_mut().insert_non_send_resource(resource);
+        self.main_mut().insert_non_send_resource(resource);
```

Comment:
I really don't want to break users like this. Why was the swap from `world_mut` to `main_mut` required as part of this PR?

Reply:
It wasn't a requirement, I made it automatically when adding the other methods. I could add this in the same "adding methods to App and SubApps PR) if you prefer.

It doesn't break users, in any case, since `main_mut` is a wrapper to `app.sub_apps.main`, and it's `insert_non_send_resource` calls it's `world`, while `world_mut` is a wrapper to `main_mut().world_mut()` (so, the same world).

Reply:
Restored and moved to [this](https://github.com/bevyengine/bevy/pull/13425/files) as well.